src/include/daos/dtx.h

```diff
@@ -46,11 +47,25 @@ struct dtx_conflict_entry {
 static inline void
 daos_dti_gen(struct dtx_id *dti, bool zero)
 {
+	static __thread struct {
```

Comment:
Since daos_dti_gen is inline, does this compile to one object for each call site or just one for all call sites?

Reply:
Ah right, should avoid to compile to multiple object...

Reply:
To be honest, I don't know what the compiler does in this case. Something to investigate.

Reply:
I am not very sure as well, by some check - "when the compiler inline-expands a function call, the function’s code gets inserted into the caller’s code stream (conceptually similar to what happens with a #define macro)", I think likely that possibly generate the obj in caller's code.

Reply:
The best might be to avoid this question by making it a global.